miio/huizuo.py

```diff
@@ -0,0 +1,154 @@
+
+
+class HuizuoException(DeviceException):
+    pass
+
+
+class HuizuoStatus:
+    def __init__(self, data):
+        """
+        Response of a Huizuo Pisces For Bedroom (huayi.light.pis123)
+         {'id': 1, 'result': [
+             {'did': '', 'siid': 2, 'piid': 1, 'code': 0, 'value': False},
+             {'did': '', 'siid': 2, 'piid': 2, 'code': 0, 'value': 94},
+             {'did': '', 'siid': 2, 'piid': 3, 'code': 0, 'value': 6400}
+             ]
+        }
+
+        Explanation (line-by-line):
+            power = '{"siid":2,"piid":1}' values = true,false
+            brightless(%) = '{"siid":2,"piid":2}' values = 1-100
+            color temperature(Kelvin) = '{"siid":2,"piid":3}' values = 3000-6400
+        """
+
+        self.data = data
+
+    @property
+    def is_on(self) -> bool:
+        """Return whether the lamp is on or off."""
+        return self.data["power"]
+
+    @property
+    def brightness(self) -> int:
+        """Return current brightness."""
+        return self.data["brightness"]
+
+    @property
+    def color_temp(self) -> int:
+        """Return current color temperature."""
+        return self.data["temperature"]
+
+    def __repr__(self):
+        s = "<Huizuo on=%s brightness=%s color_temp=%s>" % (
+            self.is_on,
+            self.brightness,
+            self.color_temp,
+        )
+        return s
+
+
+class Huizuo(Device):
+    """A support for Huizuo PIS123."""
+
+    def __init__(
+        self,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+        model: str = MODEL_HUIZUO_PIS123,
+    ) -> None:
+        super().__init__(ip, token, start_id, debug, lazy_discover)
+
+        if model in MODELS_SUPPORTED:
+            self.model = model
+        else:
+            self.model = MODEL_HUIZUO_PIS123
+            _LOGGER.error(
+                "Device model %s unsupported. Falling back to %s.", model, self.model
+            )
+
+    @command(
+        default_output=format_output(
+            "\n",
+            "Power: {result.is_on}\n"
+            "Brightness: {result.brightness}\n"
+            "Temperature: {result.color_temp}\n"
+            "\n",
+        )
+    )
+    def status(self) -> HuizuoStatus:
+        """Retrieve properties."""
+        properties = [
+            {"siid": 2, "piid": 1},
+            {"siid": 2, "piid": 2},
+            {"siid": 2, "piid": 3},
+        ]
+        properties_name = ["power", "brightness", "temperature"]
+
+        values = self.get_properties(properties)
+        values_info = []
+        for value in values:
+            values_info.append(value["value"])
+
+        return HuizuoStatus(dict(zip(properties_name, values_info)))
+
+    @command(
+        default_output=format_output("Powering on"),
+    )
+    def on(self):
+        """Power on."""
+        return self.raw_command("set_prop", [{"siid": 2, "piid": 1, "value": True}])
```

Comment:
@rytilahti The MiotDevice class implements `set_property` which isn't used here because the device expects `set_prop`? @darckly Could try to use `set_property` instead of `set_prop`. May be your device supports both.

Reply:
@syssi Thank you for the hint regarding MiotDevice class - I just checked and looks like set_property works for this lamp also. I'll rewrite tomorrow using this class and upload an update.

Reply:
Cool! I will provide some tests later the week (hopefully).

Reply:
I have uploaded changes and some tests

Reply:
Good catch!